s6/service

```diff
@@ -2,29 +2,49 @@
 # This script patches all service commands into the appropriate s6- commands
 # pi-hole upstream scripts need a 'service' interface. why not systemd? docker said so.
 start() {
-    s6-svc -wU -u -T2500 /var/run/s6/services/$service
+  s6-svc -wU -u -T2500 /var/run/s6/services/$service
 }
 
 stop() {
-    s6-svc -wD -d -T2500 /var/run/s6/services/$service
+  s6-svc -wD -d -T2500 /var/run/s6/services/$service
 }
 
 restart() {
+  local pid
+
+  # Get the PID of the service we are asking to restart
+  pid=$(pgrep $service)
+
+  # Only attempt to stop the service if it is already running
+  if [ -n "$pid" ]; then
     stop
+
+    # Loop until we are certain that the process has been stopped
+    while test -d /proc/$pid; do
+      sleep 0.2
+    done
+  fi
```

Comment:
Might want something like an upper limit on time so that you don't accidentally infinite loop yourself.

Reply:
I'm trying to think of what could go wrong to get us stuck in here - i.e FTL not stopping for whatever reason - buy you're probably right, more guards are better!